src/main/java/seedu/address/model/scheduler/Timetable.java

```diff
@@ -56,13 +58,17 @@ public void addLesson(Lesson lesson) {
      * Merges another timetable together
      */
     public void mergeTimetable(Timetable other) {
+        logger.info("=============================[ Timetable: Check conflict ]===========================");
         assert(!TimeUtil.hasConflict(this, other));
-        HashMap<Day, ArrayList<HourBlock>> schedule = other.getSchedule();
+        logger.info("=============================[ Timetable: No Conflict ]===========================");
+        HashMap<Day, ArrayList<HourBlock>> otherSchedule = other.getSchedule();
         for (Day day : Day.values()) {
-            ArrayList<HourBlock> dayEvents = schedule.get(day);
-            for (int i = 0; i < schedule.size(); i++) {
-                if (dayEvents.get(i).getCommitment().isPresent()) {
-                    Commitment lesson = dayEvents.get(i).getCommitment().get();
+            ArrayList<HourBlock> dayEvents = otherSchedule.get(day);
+            for (int i = 0; i < dayEvents.size(); i++) {
+                if (!dayEvents.get(i).isFree()) {
+                    Commitment lesson = dayEvents.get(i).getCommitment().orElseGet(() -> null);
```

Comment:
Hmm, if it is going to be retrieved as null, then what’s the point of having an optional lol?